keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -663,9 +813,13 @@ is disabled [Disconnected node when taint-manager is disabled](#disconnected-nod
 However, as disabling taint-manager is deprecated it is not a concern for this
 KEP.
 
-For Alpha, we implement a fix for this issue by setting the pod phase as `Failed`
-in podgc. For Beta, we are going to simplify the code in job controller
-responsible for detecting pod failures.
+For Alpha, we implemented a fix for this issue by setting the pod phase as `Failed`
+in podgc.
+
+For Beta, we are going to simplify the code in job controller
+responsible for detecting pod failures. For this purpose we are going to introduce
+a worker in the disruption controller to mark pods stuck in the pending phase,
```

Comment:
What cases will this cover? Can you check what happens in particular for:

- Unschedulable pods (for example, with node affinity to a node that doesn't exist).
- Unknown docker image. Once you mark the pod for deletion, would kubelet leave it in Pending phase?

Reply:
Done experiments. It turns out that the first case of terminating unchedulable pods is already covered by PodGC in this branch: https://github.com/kubernetes/kubernetes/blob/7f129f1c9af62cc3cd4f6b754dacdf5932f39d5c/pkg/controller/podgc/gc_controller.go#L289. The scheduled terminating pods which are in the pending phase currently remain in this phase. 

However, since one of the two scenarios is already covered by PodGC I think instead of introducing another disruption controller worker it will be simpler to extend PodGC. Adjusted the KEP accordingly.